### PR TITLE
Hide potentially sensitive values (system properties and environment variables) by default

### PR DESCRIPTION
**Pull request by daniel-beck**
Pages like `/systemInfo` in Jenkins show potentially sensitive information to users entitled to view them. To prevent unintentionally revealing that information to shoulder surfers or when screensharing, this PR proposes to hide the values in the tables by default, adding buttons to reveal individual items or all of them.

This is not a security fix, but a security related enhancement.

image

image

image

image

### Proposed changelog entries

* Hide values in tables showing  potentially sensitive system properties and environment variables by default.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6843"></a>

### CODE REVIEW

File: core/src/main/resources/lib/hudson/propertyTable.jelly
Comment (NotMyFault) [inline core/src/main/resources/lib/hudson/propertyTable.jelly:45]:
```suggestion
        <l:icon src="symbol-view"/>
```
Any objections about using the eye symbol? It's used in different locations to highlight something can be looked at.

Comment (daniel-beck) [reply core/src/main/resources/lib/hudson/propertyTable.jelly]: Great suggestion, thanks!

**Review by NotMyFault — APPROVED** [review]
Works out nice

**Comment by daniel-beck** [comment]
@NotMyFault Thanks! Any suggestions how I can resolve the TODO issue?

**Comment by NotMyFault** [comment]
> @NotMyFault Thanks! Any suggestions how I can resolve the TODO issue?

I have no specific pointers in mind at the moment, sorry 😢

**Comment by janfaracik** [comment]
> TODO:
> 
> * [ ]  Figure out how to overlap the individual reveal button and the text in a way compatible with localization. Right now, I just add a negative margin of fixed width.

Might be possible with CSS Grid - https://mastery.games/post/overlapping-grid-items/

**Comment by daniel-beck** [comment]
@janfaracik Thanks, that seems to work! WDYT?

**Review by timja — COMMENTED** [review]
This seems to be a lot of buttons to me which feels a bit strange

Could we use links instead?

Also would it make sense to allow the user to hide the value after looking at it?

This is how Azure looks for this if it helps:

image

image

image

image

**Comment by daniel-beck** [comment]
Looks weird, there seems to be something broken with `jenkins-link--with-icon`:

Screenshot 2022-07-22 at 22 51 29

**Comment by NotMyFault** [comment]
The grid stuff doesn't work well if you have a bunch of path variables set, e.g.:

**Comment by timja** [comment]
I have some WIP here which I'll pick back up later on:
https://github.com/daniel-beck/jenkins/pull/9

**Comment by timja** [comment]
https://github.com/daniel-beck/jenkins/pull/9 is functional now and looks quite good I think, there's a couple of CSS issues to improve in that one too described in the descroption

**Comment by timja** [comment]
I'm happy with https://github.com/daniel-beck/jenkins/pull/9 now if you want to take a look :)

**Review by timja — APPROVED** [review]
LGTM after minor fixups that I've pushed

**Comment by NotMyFault** [comment]
The value itself should probably not be a button, because that makes copying impossible, no?

**Comment by timja** [comment]
> The value itself should probably not be a button, because that makes copying impossible, no?

Copying works fine:
image

**Review by NotMyFault — APPROVED** [review]
> Copying works fine:

Huh? Then I had a bad time earlier.

**Comment by timja** [comment]
> > Copying works fine:
> 
> Huh? Then I had a bad time earlier.

You have to drag from outside of the button on the right side.

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by daniel-beck** [comment]
@timja Copy & paste of values is completely broken in Firefox. Selecting all text on the page and copying it does not copy a single value even if all are shown. Could you look into that, or consider reverting this?

**Comment by timja** [comment]
> @timja Copy & paste of values is completely broken in Firefox. Selecting all text on the page and copying it does not copy a single value even if all are shown. Could you look into that, or consider reverting this?

Looking

**Comment by timja** [comment]
Reproduced, interesting, selecting all in chrome does get all values whereas in firefox it's not selected.

Investigating

**Pull request merged**
